libs/openFrameworks/gl/ofGLProgrammableRenderer.cpp

```diff
@@ -1315,12 +1321,46 @@ void ofGLProgrammableRenderer::bind(const ofFbo & fbo, bool setupPerspective){
 }
 
 //----------------------------------------------------------
-void ofGLProgrammableRenderer::unbind(const ofFbo & fbo){
-	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+void ofGLProgrammableRenderer::end(const ofFbo & fbo){
+	fbo.unbind();
```

Comment:
mmh ok, let me take a look later. in general calling bind/unbind begin/end or draw in GL objects is now the only operations that are thread unsafe so to work with threads you should be able to do:

```
renderer.bind(fbo)
```

and that should be safe but if that internally calls fbo.bind and fbo calls ofGetGLRenderer then it'll not be safe any more. 

why do we need the frameBuffer stack? does that solve some problem? i've been also trying to avoid keeping this kind of state even inside the renderer since it's really prone to errors. it would be weird to have nested fbo binds no?